### PR TITLE
Pushing getAllProducts

### PR DESCRIPTION
**Pull request by Sach24eth**

### CODE REVIEW

File: server/routes/productRoute.js [hunk 1]
```diff
@@ -3,32 +3,71 @@ const router = express.Router();
...
-                return res.status(500).send();
-            }
-        }
+  const body = req.body;
+  console.log("body", body);
+
+  //console.log(option);
+  let searchTerm = body.search;
+  console.log(searchTerm);
+  if (!searchTerm || searchTerm.trim() == 0) {
```
Comment (leojacob7) [inline server/routes/productRoute.js:25]: I think it should be searchterm.trim().length

File: server/routes/productRoute.js [hunk 2]
```diff
@@ -3,32 +3,71 @@ const router = express.Router();
...
+          products: search_List,
+        });
+      } catch (e) {
+        return res.status(404).render("errorPage/noSearch");
+      }
+    }
+  }
+});
+
+router.post("/addProduct", async (req, res) => {
```
Comment (leojacob7) [inline server/routes/productRoute.js:58]: I think the portion to check if user is logged in is automatically handled if you include authMiddleware

File: server/views/errorPage/noSearch.handlebars [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<div class="container d-flex align-items-center flex-column">
+  <div class="scontainer d-flex align-items-center flex-column">
+    <h1>Unfortunately we did not find any products!</h1>
+    <p>We do not have any products with the description entered.</p>
+    <img
+      class="thumb-image loaded notFound"
+      data-src="https://images.squarespace-cdn.com/content/v1/51cdafc4e4b09eb676a64e68/1470175715831-NUJOMI6VW13ZNT1MI0VB/image-asset.jpeg"
```
Comment (leojacob7) [inline server/views/errorPage/noSearch.handlebars:7]: can you add a //TODO: comment to remind a clean up when we have our stuff next week?

File: server/routes/productRoute.js (hunk 2 above)
Comment (Sach24eth) [reply server/routes/productRoute.js:58]: This I had in my previous code too, This was before the addProduct was takenby Shilin. I'm not sure why this is coming up as a change, I have not started this as of yet

File: server/views/errorPage/noSearch.handlebars (hunk 3 above)
Comment (Sach24eth) [reply server/views/errorPage/noSearch.handlebars:7]: I'll add that

File: server/routes/productRoute.js (hunk 1 above)
Comment (Sach24eth) [reply server/routes/productRoute.js:25]: yes Sorry I think I missed this.

**Review by Sach24eth — COMMENTED** [review]
Made the necessary changes. I also added the search as per what we discussed today and made a completely new page for it

**Review by leojacob7 — APPROVED** [review]

**Pull request merged**